common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -116,12 +117,56 @@ public static async Task<ErrorCode> GetExceptionCodeAsync(HttpResponseMessage re
             try
             {
                 IoTHubExceptionResult responseContent = JsonConvert.DeserializeObject<IoTHubExceptionResult>(responseContentStr);
-                Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
 
-                if (messageFields != null
-                    && messageFields.TryGetValue(CommonConstants.ErrorCode, out string errorCodeObj))
+                try
                 {
-                    errorCodeValue = Convert.ToInt32(errorCodeObj, CultureInfo.InvariantCulture);
+                    Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
+
+                    if (messageFields != null
+                        && messageFields.TryGetValue(CommonConstants.ErrorCode, out string errorCodeObj))
+                    {
+                        errorCodeValue = Convert.ToInt32(errorCodeObj, CultureInfo.InvariantCulture);
+                    }
+                }
+                catch (JsonReaderException ex)
+                {
+                    if (Logging.IsEnabled)
+                        Logging.Error(null, $"Failed to deserialize error message into a dictionary: {ex}. Message body: '{responseContentStr}.'");
+
+                    // In some scenarios, the error response string is a ';' delimited string with the service-returned error code.
+                    const char errorFieldsDelimiter = ';';
+                    string[] messageFields = responseContent.Message.Split(errorFieldsDelimiter);
+
+                    if (messageFields != null)
+                    {
+                        _ = messageFields
+                            .Any(field =>
+                            {
+#if NET451 || NET472 || NETSTANDARD2_0
+                                if (field.IndexOf(CommonConstants.ErrorCode, StringComparison.OrdinalIgnoreCase) >= 0)
+#else
+                                if (field.Contains(CommonConstants.ErrorCode, StringComparison.OrdinalIgnoreCase))
+#endif
+                                {
+                                    const char errorCodeDelimiter = ':';
+                                    string[] errorCodeFields = field.Split(errorCodeDelimiter);
+
+                                    if (Enum.TryParse(errorCodeFields[1], out ErrorCode errorCode))
+                                    {
+                                        errorCodeValue = (int)errorCode;
+                                    }
+                                }
+                                return true;
```

Comment:
Should this return a status code and not true??

Reply:
No, this is the return value for the linq "Any" function, which is bool. We don't need that return value for our existing logic, so I've now removed this.